src/DependencyInjection/Configuration.php

```diff
@@ -38,6 +38,8 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->scalarNode('client')->defaultValue('httplug.client')->end()
+                        ->scalarNode('request_factory')->defaultValue('nexy_slack.request_factory.default')->end()
+                        ->scalarNode('stream_factory')->defaultValue('nexy_slack.stream_factory.default')->end()
```

Comment:
Same logic as with the HTTP client class, allow a bundle user to replace the request and stream factories with their own implementation.